tasks/deploy/greeter.ts

```diff
@@ -1,14 +1,33 @@
-import { task } from "hardhat/config";
+import { task, types } from "hardhat/config";
 import { TaskArguments } from "hardhat/types";
 
 import { Greeter } from "../../src/types/Greeter";
 import { Greeter__factory } from "../../src/types/factories/Greeter__factory";
+import { SUBTASK_DEPLOY_VERIFY, SUBTASK_DEPLOY_WAIT_FOR_CONFIRMATIONS, TASK_DEPLOY_GREETER } from "../constants";
+
+task(TASK_DEPLOY_GREETER)
+  .addParam("greeting", "Say hello, be nice", "Hello, World", types.string)
+  .addParam("confirmations", "How many block confirmations to wait for", 0, types.int)
```

Comment:
Why do we need this if we already have it in the verify task?

Reply:
This might be a bit of an overkill, but it handles the case of `--verify false` (in which case the confirmations in verify wont run), but you need to have some `--confirmations`